### PR TITLE
add layer support

### PR DESCRIPTION
**Pull request by snodnipper**
Improve layer support within the vector tile library. 

I am not precious about any of this code and I am happy to refactor as appropriate - for example, it might be helpful to rename _LayerGroup_ as _Mvt_.

Thanks!

### CODE REVIEW

**Comment by ShibaBandit** [comment]
Thank's for the pull request, I'll check this out some more soon.

File: src/test/java/com/wdtinc/mapbox_vector_tile/adapt/jts/LayerTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class LayerTest {
+
+    private static final GeometryFactory GEOMETRY_FACORY = new GeometryFactory();
```
Comment (ShibaBandit) [inline src/test/java/com/wdtinc/mapbox_vector_tile/adapt/jts/LayerTest.java]: Noticed minor typo here, 'T' in factory.

Comment (snodnipper) [reply src/test/java/com/wdtinc/mapbox_vector_tile/adapt/jts/LayerTest.java]: np - amended that.

Are you on thespatialcommunity.org's slack channel?  If not it is easy to sign up and we can chat easily about various possible improvements.

**Comment by ShibaBandit** [comment]
I'm seeing the needs behind this pull request now. It may not be possible to get the MVT metadata like the layer name + other metadata back out of a JTS tile layers. I'm not sure I understand the 'Sink' private interface change yet. Using a private interface in a public API method likely an issue.  I may provide some edits if that's alright.

**Comment by ShibaBandit** [comment]
Is the use of a Sink an attempt to mix in some reactive (Rx) programming style? Maybe this could be taken out and put into a separate pull request. I have some concerns about it, especially where every geometry involves a layer name lookup (linear search layer name string comparison?).

**Comment by ShibaBandit** [comment]
One super picky comment - I generally format code with IntelliJ - auto-indent so function parameters line up. Some new functions in MVTReader have misaligned parameters.

**Comment by snodnipper** [comment]
thanks for the comments.

1) Sink interface - I only created this to reduce code duplication and maintain existing code contracts.  There are two slightly different code paths that could be simplified (but would be a breaking change).  

[`loadMvt(is, geomFactory, tagConverter, ringClassifier, (layerName, geometry) -> tileGeoms.add(geometry));`](https://github.com/OrdnanceSurvey/mapbox-vector-tile-java/blob/3a7d912ba32ed93c8c797829eb7a4adabf00f54b/src/main/java/com/wdtinc/mapbox_vector_tile/adapt/jts/MvtReader.java#L117)

and

[`loadMvt(is, geomFactory, tagConverter, ringClassifier, (layerName, geometry) -> {
            Layer layer = layerGroup.getLayer(layerName);
            if (layer == null) {
                layer = new Layer(layerName);
                layerGroup.addLayer(layer);
            }
            layer.add(geometry);
        });`](https://github.com/OrdnanceSurvey/mapbox-vector-tile-java/blob/3a7d912ba32ed93c8c797829eb7a4adabf00f54b/src/main/java/com/wdtinc/mapbox_vector_tile/adapt/jts/MvtReader.java#L260)

imho, that complexity could be designed out by writing `loadMvt` to only use Layers...the code block is:
```java
        for(VectorTile.Tile.Layer nextLayer : mvt.getLayersList()) {
            String layerName = nextLayer.getName();
```
We wouldn't therefore need to continually perform lookups for the layer.

_I agree that the public / private aspect of `void loadMvt` was confusing - I have now marked it as private._  

2) Code formatting - no problem, I hope to look at that tomorrow.  I normally use [checkstyle rules](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml) and have been working with IntelliJ in a completely unknown state.

I have added you to our repo should you want to commit further on this branch.  No problem either way _but_ I can stop amending commits if we are both working on the branch.

**Comment by snodnipper** [comment]
I am currently working on a "feature/layer-enhancement" branch to deal with the inefficiencies discussed - I will message when uploaded for further discussion.

**Comment by ShibaBandit** [comment]
See suggested edits at https://github.com/OrdnanceSurvey/mapbox-vector-tile-java/pull/1

**Comment by snodnipper** [comment]
PR has latest code (squashed).  I am remote today (limited battery / network) so if you'd like to keep more history then I'll have to deal with that tomorrow.

**Comment by ShibaBandit** [comment]
Merged, thank you for your contributions. Will look into maven central release jar very soon with new 2.0.0 version.

**Comment by ShibaBandit** [comment]
Available in 2.0.0 in maven central. @snodnipper

**Pull request merged**